src/Avalonia.Base/Platform/IAssetLoader.cs

```diff
@@ -77,5 +77,11 @@ public interface IAssetLoader
         /// <param name="baseUri">The base URI.</param>
         /// <returns>All matching assets as a tuple of the absolute path to the asset and the assembly containing the asset</returns>
         IEnumerable<Uri> GetAssets(Uri uri, Uri? baseUri);
+        
+        /// <summary>
+        /// Removes the assembly from the cache.
+        /// </summary>
+        /// <param name="name">The Assemblies.First().GetName().Name</param>
+        void RemoveAssembly(string name);
```

Comment:
You probably also want to add this method to https://github.com/AvaloniaUI/Avalonia/blob/a37de40867af70ddaefc2a2d087919d2f9903c4b/src/Avalonia.Base/Platform/AssetLoader.cs static class.
As users normally use that static class instead of interface directly.

Also, wouldn't API like `void InvalidateAssemblyCache()` be more useful in general here?